*** Message.pm.orig	2010-09-10 02:02:45.000000000 +0200
--- Message.pm	2023-06-14 13:20:40.053315612 +0200
***************
*** 21,26 ****
--- 21,29 ----
  
  use Math::BigInt();
  
+ # Added by NeDi:
+ use Time::HiRes;
+ 
  ## Version of the Net::SNMP::Message module
  
  our $VERSION = v3.0.1;
***************
*** 685,690 ****
--- 688,696 ----
     }
  
     DEBUG_INFO('transport address %s', $this->{_transport}->dest_taddress());
+    # Added by NeDi
+    $this->{_transport}->{_send_time} = Time::HiRes::time;
+ 
     $this->_buffer_dump();
  
     if (defined (my $bytes = $this->{_transport}->send($this->{_buffer}))) {
